WPXMLRPC/WPXMLRPCDataCleaner.m

```diff
@@ -191,30 +193,77 @@ - (NSString *)cleanWithTidyIfPresent:(NSString *)str {
  the length of these extra leading characters.
  Check for a good closing tag, and try to repair a broken closing tag.
  */
-- (NSString *)cleanClosingTagIfNeeded:(NSString *)str
+- (NSString *)cleanClosingTagIfNeeded:(NSString *)str lengthOfCharactersPrecedingPreamble:(NSInteger)length
 {
+    // Clean responses, but not requests (for now).
+    if ([str rangeOfString:@"methodResponse"].location == NSNotFound) {
```

Comment:
This is a purely style comment.  I'm being a bit nitpicky, so feel free to disagree.

I couldn't find any obvious logic mistakes in this code.  However one thing that I did notice when reading this method is that following the logic is a bit difficult.  I feel this is backed up a bit by the amount of comments in the code.

There are two suggestions I can make, to make this method slightly easier to follow:

1. Add some extra `BOOL`s to make the conditions being checked more obvious to the reader.  For example instead of:

```objective-c
if ([str rangeOfString:@"methodResponse"].location == NSNotFound) {
```

Use:

```objective-c
BOOL strContainsAResponse = ([str rangeOfString:@"methodResponse"].location != NSNotFound);

if (!strContainsAResponse) {
```

2. Break this method up into sub methods so that instead of having these lines inline...

```objective-c
NSString *closingTags;
if ([str rangeOfString:@"<params>"].location == NSNotFound) {
    closingTags = @"</value></fault></methodResponse>";
} else {
    closingTags = @"</param></params></methodResponse>";
}
```

...we have...

```objective-c
NSString *closingTags = [self closingTagsForRequest:str];
```